src/core/document.js

```diff
@@ -62,33 +62,16 @@ var Page = (function PageClosure() {
   }
 
   Page.prototype = {
-    getInheritedPageProp: function Page_getInheritedPageProp(key, getArray) {
-      var dict = this.pageDict, valueArray = null, loopCount = 0;
-      var MAX_LOOP_COUNT = 100;
-      getArray = getArray || false;
-      // Always walk up the entire parent chain, to be able to find
-      // e.g. \Resources placed on multiple levels of the tree.
-      while (dict) {
-        var value = getArray ? dict.getArray(key) : dict.get(key);
-        if (value !== undefined) {
-          if (!valueArray) {
-            valueArray = [];
-          }
-          valueArray.push(value);
-        }
-        if (++loopCount > MAX_LOOP_COUNT) {
-          warn('getInheritedPageProp: maximum loop count exceeded for ' + key);
-          return valueArray ? valueArray[0] : undefined;
-        }
-        dict = dict.get('Parent');
-      }
-      if (!valueArray) {
-        return undefined;
-      }
-      if (valueArray.length === 1 || !isDict(valueArray[0])) {
-        return valueArray[0];
+    /**
+     * @private
+     */
+    _getInheritableProperty(key, getArray = false) {
+      let value = getInheritableProperty(this.pageDict, key, getArray,
+                                         /* stopWhenFound = */ false);
+      if (!Array.isArray(value)) {
+        return value;
       }
-      return Dict.merge(this.xref, valueArray);
+      return isDict(value[0]) ? Dict.merge(this.xref, value) : value[0];
```

Comment:
Please note that `Dict.merge` is meaningless, and will only slow things down (albeit probably not much), when `value.length === 1`.

Reply:
Done in the new commit.